[AD-4057] React config

johndewolf opened the pull request:
## What does this PR do?

- Add config for React linting

## Resources

- [Jira ticket](https://getpantheon.atlassian.net/browse/AD-4057)

johndewolf commented on package.json:
```diff
@@ -3,23 +3,49 @@
-    "@babel/eslint-parser": "^7.16.3",
-    "@babel/preset-env": "^7.16.4",
-    "babel-plugin-module-resolver": "^4.1.0",
+    "babel-eslint": "~10.1.0",
```
Removed babel-eslint

pantheon-releases commented:
<!--
  1 failure:  Danger failed to ...
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Danger failed to run `danger.config.js`.
  </td>
    </tr>
  </tbody>
</table>



## Error SyntaxError
```
Cannot use import statement outside a module
danger.config.js:2
import { gt } from 'semver';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:979:16)
    at Module._compile (internal/modules/cjs/loader.js:1027:27)
    at Object.requireFromString [as default] (/root/project/node_modules/require-from-string/index.js:28:4)
    at /root/project/node_modules/danger/distribution/runner/runners/inline.js:157:63
    at step (/root/project/node_modules/danger/distribution/runner/runners/inline.js:52:23)
    at Object.next (/root/project/node_modules/danger/distribution/runner/runners/inline.js:33:53)
    at /root/project/node_modules/danger/distribution/runner/runners/inline.js:27:71
    at new Promise (<anonymous>)
    at __awaiter (/root/project/node_modules/danger/distribution/runner/runners/inline.js:23:12)
    at exports.runDangerfileEnvironment (/root/project/node_modules/danger/distribution/runner/runners/inline.js:118:136)
```
### Dangerfile
```
-------------------^
```
  
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 93a8931983c7dcc4a00f13999db069632bf1e133
</p>

puglyfe commented:
I think the danger issue is because we moved the babel config into eslint. so now the root project doesn't have any babel config. we may need to add it to package.json or convert the dangerfile to non-module syntax.

johndewolf commented:
> I think the danger issue is because we moved the babel config into eslint. so now the root project doesn't have any babel config. we may need to add it to package.json or convert the dangerfile to non-module syntax.

Looks like moving babel config to package.json fixed it?

puglyfe commented:
> > I think the danger issue is because we moved the babel config into eslint. so now the root project doesn't have any babel config. we may need to add it to package.json or convert the dangerfile to non-module syntax.
> 
> Looks like moving babel config to package.json fixed it?

i think we’ll need it in both places. package.json will only apply to this repo not the consumer

johndewolf commented:
> > > I think the danger issue is because we moved the babel config into eslint. so now the root project doesn't have any babel config. we may need to add it to package.json or convert the dangerfile to non-module syntax.
> > 
> > 
> > Looks like moving babel config to package.json fixed it?
> 
> i think we’ll need it in both places. package.json will only apply to this repo not the consumer

Ah i was wondering that, updated
